### PR TITLE
Add API check whether option in getOptionInfo() exists

### PR DESCRIPTION
**Pull request by nafur**
This PR adds a missing check in the API for `getOptionInfo()`.

### CODE REVIEW

File: test/unit/api/solver_black.cpp [hunk 1]
```diff
@@ -1332,10 +1332,7 @@ TEST_F(TestApiBlackSolver, getOptionNames)
 TEST_F(TestApiBlackSolver, getOptionInfo)
 {
   {
-    auto info = d_solver.getOptionInfo("verbose");
-    ASSERT_EQ(info.name, "verbose");
-    ASSERT_EQ(info.aliases, std::vector<std::string>{});
-    ASSERT_TRUE(std::holds_alternative<api::OptionInfo::VoidInfo>(info.valueInfo));
+    EXPECT_THROW(d_solver.getOptionInfo("asdf-invalid"), CVC5ApiException);
```
Comment (4tXJ7f) [inline test/unit/api/solver_black.cpp:1335]: Why not keep both tests?

**Review by 4tXJ7f — CHANGES_REQUESTED** [review]
LGTM, except for a minor comment.

File: test/unit/api/solver_black.cpp (hunk 1 above)
Comment (nafur) [reply test/unit/api/solver_black.cpp:1335]: Changed the behaviour after some offline discussion, now both tests are kept.

**Review by mpreiner — APPROVED** [review]

**Review by 4tXJ7f — APPROVED** [review]
LGTM, thanks for the update.

**Pull request merged**